src/cmd.rs

```diff
@@ -125,13 +124,27 @@ pub fn exec(
     // stderr should only ever contain error+ level diagnostics
     if code != 0 {
         let messages: Vec<_> = match String::from_utf8(output.stderr) {
-            Ok(errors) => errors
-                .lines()
-                .filter_map(crate::error::Message::parse)
-                .collect(),
-            Err(e) => vec![Message::fatal(format!(
-                "unable to read stderr ({}) but process exited with code {}",
-                e, code
+            Ok(errors) => {
+                let mut messages = Vec::new();
+
+                for line in errors.lines() {
+                    if let Some(msg) = crate::error::Message::parse(line) {
+                        messages.push(msg);
+                    } else if let Some(msg) = messages.last_mut() {
+                        if !line.is_empty() {
```

Comment:
Empty lines might have (stylistic) relevance, maybe just add this kind of post-processing (and not skip them)?
```rust
for msg in &mut messages {
    msg.notes.truncate(msg.notes.trim_end_matches('\n').len());
}